rules/rules-reviewed/eap8/eap6/eap8.ejb.windup.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="eap8-ejb"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides analysis of removed EJB methods in JBoss EAP 8.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="[6,7)" />
+        <targetTechnology id="eap" versionRange="[8,)" />
+        <tag>eap8</tag>
+    </metadata>
+    <rules>
+        <rule id="eap8-ejb-00001">
+            <when>
```

Comment:
According to the comment on the ticket:
> So the rules need to look for method calls below, on a class that implements the above interface.

I think the rules are missing the `on a class that implements the above interface` part. So I think it should be something like this:

```
                <javaclass references="EJBContext">
                    <location>IMPLEMENTS</location>
                </javaclass>
                <javaclass references="javax.ejb.EJBContext.getCallerIdentity({*})" from="default">
                    <location>METHOD_CALL</location>
                </javaclass>
```

Reply:
The ticket also mentions this
> WINDUP's ([Javadoc](http://windup.github.io/windup/docs/latest/javadoc/org/jboss/windup/ast/java/data/TypeReferenceLocation.html)) METHOD_CALL location qualifier works transitively with interfaces. So the rules can be coded to look for
>FullyQualifiedInterfaceName.MethodName()
>And any concrete classes that have implemented the interface and calling the method will be identified.

So I don't think we need to make those changes. wdyt?

Reply:
Isn't it possible that the METHOD_CALL is being made from outside the class, even with a fully qualified name?

Reply:
I thought the point made in the ticket was that the METHOD_CALL qualifier will pick up calls on classes implementing the interface too -  there's no other way of calling these methods is there?